R/Reactant/build_tarballs.jl

```diff
@@ -450,7 +463,7 @@ for (i,build) in enumerate(builds)
     build_tarballs(i == lastindex(builds) ? non_platform_ARGS : non_reg_ARGS,
                    name, version, build.sources, build.script,
                    build.platforms, build.products, build.dependencies;
-                   preferred_gcc_version=v"10", preferred_llvm_version=v"18", julia_compat="1.6",
-                   augment_platform_block, lazy_artifacts=true)
+                   preferred_gcc_version=build.preferred_gcc_version, preferred_llvm_version=v"18", julia_compat=build.julia_compat,
+                   augment_platform_block, lazy_artifacts=true, lock_microarchitecture=false)
```

Comment:
`Reactant.jl` requires Julia v1.10 anyway: https://github.com/EnzymeAD/Reactant.jl/blob/eeaf86c0555236c238729805a29ee1435f870450/Project.toml#L67
```suggestion
                   preferred_gcc_version=build.preferred_gcc_version, preferred_llvm_version=v"18", julia_compat="1.10",
                   # We use GCC 13, so we can't dlopen the library during audit
                   augment_platform_block, lazy_artifacts=true, lock_microarchitecture=false, dont_dlopen=true)
```